### DISCUSSION TITLE
Round QR SVG Render

### DISCUSSION THREAD

**Discussion by sogorman** [chillerlan/php-qrcode#137] [General]
Good evening! I wanted to check with the group and see if this would be beneficial to anyone else.

We are looking at coming up with some SVG QR codes in a circle format. The thought is to give the code a little different aesthetic which may draw someone's eye in and increase interactivity.

Here is a mockup of what we are looking at. 

![Untitled-4](https://user-images.githubusercontent.com/40614647/177880350-944aa99e-67a0-48a2-ab4c-f24bb08ee767.jpg)

At a high level, we are thinking a SVG QR code but taking that valid code and surrounding it with what is essentially stylized 'noise' while leaving a 1 square width around the code in the quiet zone. We know that this may yield a less optimal reading but in our testing it's minimal at worst.

**Comment by codemasher** [comment]
This can be achieved by adding an oversized quiet zone via the standard options `QROptions::$quietzoneSize`, then iterate over the matrix and randomly change modules that match `QRMatrix::M_QUIETZONE` and that are within the given shape to dark. I'll post an example tomorrow as it is already sleepy time over here.

**Reply by codemasher** [reply]
Ok, it took me a bit, but how about this (example code can be found [over here](https://github.com/chillerlan/php-qrcode/blob/main/examples/svgRoundQuietzone.php)):

![img](https://user-images.githubusercontent.com/592497/178122762-1abe19ef-b23f-4252-b093-88bd4bdb79c9.svg)

**Reply by git-pranayshah** [reply]
Do you have a code of this?

**Reply by codemasher** [reply]
It is linked in the post you replied to.
